okex-client/src/client/mod.rs

```diff
@@ -34,13 +34,40 @@ pub struct OkexClient {
 }
 
 impl OkexClient {
-    pub fn new(config: OkexClientConfig) -> Self {
+    fn new(config: OkexClientConfig) -> Self {
         Self {
             client: ReqwestClient::new(),
             config,
         }
     }
 
+    pub async fn create(config: OkexClientConfig) -> Result<Self, OkexClientError> {
+        // 1. Get account configuration
+        let client = Self::new(config);
+        let path = "/api/v5/account/config";
+        let config_url = Self::url_for_path(path);
+        let headers = client.get_request_headers(path)?;
+
+        let response = client
+            .rate_limit_client(path)
+            .await
+            .get(config_url)
+            .headers(headers)
+            .send()
+            .await?;
+        let config_data =
+            Self::extract_response_data::<OkexAccountConfigurationData>(response).await?;
+
+        // 2. Check position_mode, i.e. order placement mode
+        if config_data.pos_mode == *"net_mode" {
+            return Ok(client);
+        }
+        Err(OkexClientError::PositionMode {
```

Comment:
Can we call this `MissconfiguredAccount` or something like that instead of `PositionMode`?

Also shouldn't have `msg, code` format as that is used for the responses we pass on that come directly from the API.
Perhaps just `MissconfiguredAccount(string)` or something like that.